### ISSUE TITLE
Make S2 site scheme pages responsive/Foundation-ized

### ISSUE THREAD

**Issue by nfagerlund** [dreamwidth/dw-free#2427]
Most site scheme pages are either BML (old, nonresponsive) or controller+template (new, usually converted to Foundation styling). To make a "normal" site page responsive, you convert its implementation to controller+TT, throw in a one-liner, and clean up any resulting shrapnel. (At least, that's what I'm led to believe.) 

HOWEVER, there's another class of site scheme pages that _pass through S2 instead of BML_ (styles/siteviews/layout.s2) and which act like a weird hybrid between user-styled journal pages and "normal" site scheme pages. The ones I know of are: 

- Entry page (click "reload in: site" in control strip, or disable "entry pages in my style")
- Reply page AKA slowreply (same as above)
- Icons page (always, unless the option to use your style is enabled. ...Does anyone use that? I always forget about it.)
- Tags, year-archive, month-subjects (can't ever be reached via links, but if you paste `?style=site` in the url bar, they appear to work perfectly fine)
- Reading, Network, Recent (same as previous, except they actually do look suuuuper broken)
- ~Profile??? Or does that go through BML???? Hard to tell!~ The profile page seems to be normal BML. IMO it should be high-priority because it's high-profile, but it's not part of this issue!

None of these pages use Foundation, and the usual conversion workflow doesn't apply. But the entry page and icons page in particular are probably in at least the top five high-profile nonresponsive site scheme pages remaining, so there's a big win to be had by making them work right on a dang telephone. 

The terrain is weird, but I've made some tentative progress on this. More to come. 

### WIP branch

https://github.com/nfagerlund/dw-free/tree/apr19_sitescheme_hell

### Dupes

- #1698 is a special sub-case of this issue. 

### Known blockers or semi-blockers

- [x] #2434 (midstream jquery: actual blocker, qr breaks if you don't do this)
- [ ] #2432 (adjustable comment indents: if we have to redesign the entry page from scratch anyway, let's not do anything we'll immediately want to undo)
- [x] #2411 (the "old new" userpic browser loses all its styling once you go to Foundation land, and I don't want to waste time re-styling it if its days are numbered anyway)
- [x] #2437 (extract talkform: I know this is going to get in the way at some point, I just KNOW it)

**Comment by rahaeli** [comment]
You are amazing for even considering tackling this -- we kind of all looked at it and went "arrrgh".

The historical reasons for why it works like this: those pages can all be either site-skinned or journal-skinned, depending on user settings. It used to be that the site-skinned pages were a separate BML page entirely, while journal-skinned pages went through S2, which meant the two versions got out of sync very easily and that not all styles supported all new features (which is still the case on LJ today, I believe). Making it all S2 under the hood meant you only had to update it once, even if it meant hacking together a site skin S2 style. (Say that ten times fast.)

**Comment by nfagerlund** [comment]
Okay, so, uh,

(click for full page scroll views) 

[<img src="https://user-images.githubusercontent.com/484309/57009092-68003580-6ba9-11e9-9266-69ccaa1d4b24.png" height="300">](https://user-images.githubusercontent.com/484309/57009094-6898cc00-6ba9-11e9-8c70-f2570974dd02.png) [<img src="https://user-images.githubusercontent.com/484309/57009093-6898cc00-6ba9-11e9-9c4f-b393adafd82e.png" width="300">](https://user-images.githubusercontent.com/484309/57009095-6898cc00-6ba9-11e9-96bb-480bcff6aff2.png)

This is more than a small bit janky, but it's at a point (on this one page) where _maybe it's not more broken than what's live in prod._

**Comment by nfagerlund** [comment]
Things I do not know:

- wtf is up with that first commit I made. Actually, I think the thing to do is switch resource groups before you enter the S2 code, instead of exposing a function, but idk where to do that yet. 
- How to do CSS for the entry page in a way that works well with foundation. atm I'm putting it in an SCSS file that gets imported into _skin-colors.scss, because that's the first spot I saw where I was like "ok I THIIIINK all the necessary variables will be in scope by this point on all skins," but I'm pretty sure that's not right. Also it gets loaded for every site skin page that way, and it's only used here on the entry page. 
- What colors do we use for these journal-like elements, like the alternating header backgrounds? 
- Instead of just CSSing the markup the s2 layout gave me, should I be rewriting the way it outputs comments and entries to make it work well with the pre-existing work on foundation styling? Like, maybe even use existing components? (I don't think there's many journal-like ones tho.)
- The stuff I've done so far was just trying to match the existing site scheme entry pages. But good lord, that's ten years old and plus some (since it looks a lot like the old LJ site comment pages). Maybe I shouldn't be lovingly copying this old heap?? What do we actually WANT the entry pages to look like, in a perfect world where we're not starting with all this baggage? (Or, I mean, where we're starting with maybe 20% _less_ of all this baggage. Realism.)

**Comment by rahaeli** [comment]
I missed answering these questions when you first posted them, sorry! Or at least the ones I can answer...

-- I used to know where the alternating headers were defined in site skins, but then we TTized everything and now I AM VERY CONFUSED, lol. But the CSS is .comment-wrapper-odd and .comment-wrapper-even (applied in styles/core2.s2) and you can use those classes and it should Just Work in all the skins.

-- If you want to Foundationize it, please do! IIRC we get a lot of responsiveness just from the Foundation conversion, right? But don't let that be a blocker if you would rather not.

-- I would LOVE to redesign the site skin entry pages. (I would love to redesign a LOT OF THINGS, ngl.) But I have no real idea about what it should look like other than "not that", and the entry page gets a LOT of eyeballs and people have Opinions on it, so I don't know which you'd prefer, lovingly copying that old heap or doing something different and taking it through 2-3 rounds of iteration via beta features (which maybe we should do anyway when you do this, since it's a page with a lot of eyeballs and no CSS ever survives contact with the real world of user devices intact? Or make it a new skin, Tropo Nouveau or something, to let people test that way? Though this is probably going to be markup changes, not skin changes, yeah?) 

That having been said:

* I would love to get rid of the .75em declaration in Tropo. It's for all site text, I believe, not just entry pages, but that was a decision made by one of the early devs doing the initial "make a site skin from these pretty pictures of the design". I know they had a reason for doing it, but I don't remember what it was other than the "Windows and MacOS use different definitions of what an em is and this was a compromise" that immediately jumps to mind. (I'd also like to get rid of the CSS reset in Tropo, but that's a whole 'nother ball of wax that is Outside the Scope of This Issue.)

* I am someone who finds site skinned comment pages readable and journal skinned comment pages completely unreadable, and I can't articulate why very well. Which is not very helpful, I know. :(  Part of it is I find journal skinned comment pages really visually 'busy' (even when the comment divs aren't outlined, but twice as badly when they are; it's also got a lot to do with wanting it to be easy to tell whether I'm looking at a comment page or journal view) and part of it is not being able to easily parse threading and thread expansion when threads collapse. Threaded/nested comments are one of our big differences from other sites, and our users tend to be here because they really like threaded conversation. So I'd want Entry Page 2.0 to keep the open side margins and indentations and all, but reasonably tight vertical margins between comments so it's easier to keep track of the threading. (I think "large vertical margin" is one of the things that makes me find threads in journal skinned pages difficult to follow.)

* Another reason I like site skinned entry pages is because journal skinned entry pages tend to throw way too much extraneous info on the page, like modules and the like, which (because their placement and contents vary from journal to journal) are harder for me to visually tune out than the sidebar on skins like Celerity and Gradation Vertical. (I mostly use Tropo with its complete lack of sidebar, but I use Celerity on a few alternate accounts for the visual cue to which account I'm logged into, and I can tune the sidebar out way better than I can tune journal modules out.) So I guess what I'm really saying is, site skinned entry pages should be as consistent from journal to journal as possible and not depend on the settings/input someone's picked for their journal, since site-skinned comment pages are used by people who tend to need consistency.

* It probably goes without saying, but: needs a non-JS fallback. I actually *almost* like the current LJ entry page design except for a few things that have "does not work very well AT ALL with JS off" and "expanding collapsed conversations goes way too slowly because of all the loading" at the top of the list.

* The other things I hate about LJ's current entry page, in case they help, are mostly having to do with collapsed comments: there's no way to click on a collapsed comment to dig into the threadview with that comment as the new threadroot; there's no visual indicator that draws the eye across the line the way we separate subject/poster/time/expand link with " - "; they use relative time and not absolute time. 

* DW user solarbird has been working on a more mobile-friendly style for a while (https://dreamwidthlayouts.dreamwidth.org/49147.html, https://solarbird.dreamwidth.org/1652388.html) that might be useful. It's been a while since I talked to them about it (I dropped the ball because of health stuff) but you might want to get together with them, because I know they did a lot of work to improve the visuals, not just make it more mobile-friendly.

Most of this is probably not helpful :( Maybe we should contract a designer for this, since it's SO core to the site?

**Comment by nfagerlund** [comment]
> Or make it a new skin, Tropo Nouveau or something, to let people test that way? Though this is probably going to be markup changes, not skin changes, yeah?)

Yeah, my first hope was a separate skin, but the changes are too deep. So, beta feature.

> But the CSS is .comment-wrapper-odd and .comment-wrapper-even...

Oh, no, I had that working, I just didn't know which SCSS variables I should be using for each type's bg color, to make sure I'm getting the appropriate contrast in the alternation in all of the site schemes. There weren't any named variables that fit what I was trying to do, and I was leery of using one that was completely unrelated (like "lighter border" or something random) for the alternate color. 

> solarbird stuff

Oh wow, it looks like they're doing a lot of stuff that's a lot like what I was doing to Practicality for the responsive comment indent preview. That's at a level of remove from the site skin stuff, but still sort of a prerequisite. I'll take a look later at what they're doing and see if I should bring them in on that.

**Comment by nfagerlund** [comment]
You know, the more I think about this, the more I think I was wrong about the prerequisites. 

I _personally_ don't want to use the site skin comment pages until they get a mobile redesign and something better to show threading than 25px indents. But that doesn't mean there's no value to doing a spackle job today! Especially if we do it as a beta feature. And doing a spackle job doesn't do anything to sabotage a larger redesign down the road; it's just some CSS to remove. 

I think this'll be the next thing I tackle. That Android Firefox thing was _so_ insulting. >:|
